internal/routes/oidc.go

```diff
@@ -0,0 +1,79 @@
+package routes
+
+import (
+	"net/http"
+	"net/url"
+
+	"github.com/gin-gonic/gin"
+	"go.uber.org/zap"
+)
+
+type oidcHandler struct {
+	logger *zap.SugaredLogger
+	issuer string
+}
+
+type providerJSON struct {
+	Issuer      string `json:"issuer"`
+	AuthURL     string `json:"authorization_endpoint,omitempty"`
+	TokenURL    string `json:"token_endpoint"`
+	JWKSURL     string `json:"jwks_uri"`
+	UserInfoURL string `json:"userinfo_endpoint"`
+}
+
+// Handle processes the request for the OIDC handler.
+func (h *oidcHandler) Handle(ctx *gin.Context) {
+	out := providerJSON{
+		Issuer:      h.issuer,
+		TokenURL:    buildURL(ctx, "../../token").String(),
+		JWKSURL:     buildURL(ctx, "../../jwks.json").String(),
+		UserInfoURL: buildURL(ctx, "../../userinfo").String(),
+	}
+
+	ctx.JSON(http.StatusOK, out)
+}
+
+// buildURL returns a new *url.URL for the current page being requested, overwriting the values with the ones provided.
+//
+//nolint:cyclop // necessary complexity.
+func buildURL(c *gin.Context, path string) *url.URL {
+	outURL := new(url.URL)
+
+	if c != nil && c.Request != nil && c.Request.URL != nil {
```

Comment:
You could probably reduce complexity (or the level of nested code) by inverting this logic. That is:

```
if c == nil || c.Request == nil || c.Request.URL == nil {
    return outURL
}

outURL.Path = c.Request.URL.JoinPath(path).Path
...
```

Reply:
good point, fixed.